pkg/codegen/dotnet/gen.go

```diff
@@ -1249,7 +1229,7 @@ func (mod *modContext) genType(w io.Writer, obj *schema.ObjectType, propertyType
 
 	if input {
 		pt.baseClass = "ResourceArgs"
-		if mod.details(obj).functionType {
+		if !args && mod.details(obj).plainType {
```

Comment:
Something still isn't quite right with the .NET codegen...

This schema:

```json
{
    "name": "apigateway",
    "types": {
        "apigateway:index:EventHandlerRoute": {
            "type": "object",
            "properties": {
                "path": {
                    "type": "string"
                },
                "method": {
                    "type": "string"
                },
                "function": {
                    "$ref": "/aws/v3.30.0/schema.json#/resources/aws:lambda%2Ffunction:Function"
                }
            }
        }
    },
    "resources": {
        "apigateway:index:RestAPI": {
            "isComponent": true,
            "inputProperties": {
                "routes": {
                    "type": "array",
                    "items": {
                        "$ref": "#/types/apigateway:index:EventHandlerRoute"
                    }
                }
            },
            "requiredInputs": [
                "routes"
            ],
            "plainInputs": [
                "routes"
            ],
            "properties": {
                "url": {
                    "type": "string"
                },
                "api": {
                    "$ref": "/aws/v3.30.0/schema.json#/resources/aws:apigateway%2FrestApi:RestApi"
                }
            },
            "required": [
                "url",
                "api"
            ]
        }
    },
    "language": {
        "csharp": {
            "packageReferences": {
                "Pulumi": "2.*",
                "Pulumi.Aws": "3.*"
            }
        },
        "go": {
            "generateResourceContainerTypes": true,
            "importBasePath": "github.com/pulumi/pulumi-apigateway/sdk/go/apigateway"
        },
        "nodejs": {
            "dependencies": {
                "@pulumi/aws": "^3.30.0"
            },
            "devDependencies": {
                "typescript": "^3.7.0"
            }
        },
        "python": {
            "requires": {
                "pulumi": ">=2.21.2,<3.0.0",
                "pulumi-aws": ">=3.30.0,<4.0.0"
            }
        }
    }
}
```

Is producing...

`sdk/dotnet/Inputs/EventHandlerRouteArgs.cs`:

```c#
namespace Pulumi.Apigateway.Inputs
{
    public sealed class EventHandlerRoute : Pulumi.InvokeArgs
    {
        [Input("function")]
        public Pulumi.Aws.Lambda.Function? Function { get; set; }

        [Input("method")]
        public string? Method { get; set; }

        [Input("path")]
        public string? Path { get; set; }

        public EventHandlerRoute()
        {
        }
    }
}
```

`sdk/dotnet/RestAPI.cs`:

```c#
namespace Pulumi.Apigateway
{
    public sealed class RestAPIArgs : Pulumi.ResourceArgs
    {
        [Input("routes", required: true)]
        private ImmutableArray<Inputs.EventHandlerRouteArgs>? _routes;
        public ImmutableArray<Inputs.EventHandlerRouteArgs> Routes
        {
            get => _routes ?? (_routes = new ImmutableArray<Inputs.EventHandlerRouteArgs>());
            set => _routes = value;
        }

        public RestAPIArgs()
        {
        }
    }
}
```

- Why is `EventHandlerRoute` inheriting from `InvokeArgs` and not `ResourceArgs`?
- `RestAPIArgs` is referring to `Inputs.EventHandlerRouteArgs` which doesn't exist.

Reply:
Also, nit: If we're generating a `EventHandlerRoute` type, I would expect it to be generated in `EventHandlerRoute.cs` not `EventHandlerRouteArgs.cs`.

Reply:
> RestAPIArgs is referring to Inputs.EventHandlerRouteArgs which doesn't exist.

This should be fixed with the latest changes.